pymove/visualization/folium.py

```diff
@@ -1468,3 +1437,766 @@ def plot_stops(
+        Event color.
+    user_id: String, optional, default 'id'.
+        User id column name.
+    user_point: String, optional, default 'orange'.
+        User point color.
+    line_color: String, optional, default 'blue'.
+        Line color.
+    slice_envet_show: int, optional, default None.
+
+    slice_subject_show: int, optional, default
+    """
+
+    # building structure for deltas
+    delta_event = pd.to_timedelta(window_time_event, unit='s')
+    delta_tnz = pd.to_timedelta(window_time_subject, unit='s')
+
+    # length of df_tnz
+    len_df_tnz = move_data.shape[0]
+
+    # building structure for lat and lon array
+    lat_arr = np.zeros(len_df_tnz)
+    lon_arr = np.zeros(len_df_tnz)
+
+    # folium map list
+    folium_maps = []
+
+    # for each cvp in df_cvp
+    for _, line in df_event.iterrows():
+
+        event_lat = line[event_lat_]
+
+        event_lon = line[event_lon_]
+
+        event_datetime = line[event_datetime_]
+
+        event_id = line[event_id_]
+
+        # building time window for cvp search
+        start_time = pd.to_datetime(event_datetime - delta_event)
+        end_time = pd.to_datetime(event_datetime + delta_event)
+
+        # filtering df_ for time window
+        df_filted = filters.by_datetime(
+            move_data,
+            start_datetime=start_time,
+            end_datetime=end_time
+        )
+
+        # length of df_temp
+        len_df_temp = df_filted.shape[0]
+
+        # using the util part of the array for haversine function
+        lat_arr[:len_df_temp] = event_lat
+        lon_arr[:len_df_temp] = event_lon
+
+        # building distances to cvp column
+        df_filted['distances'] = distances.haversine(
+            lat_arr[:len_df_temp],
+            lon_arr[:len_df_temp],
+            df_filted[user_lat].values,
+            df_filted[user_lon].values
+        )
+
+        # building nearby column
+        df_filted['nearby'] = df_filted['distances'].map(lambda x: (x <= radius))
+
+        # if any data for df_ in cvp time window is True
+        if df_filted['nearby'].any():
+
+            # buildng the df for the first tnz_points of tnz in nearby cvp
+            df_begin = df_filted[df_filted['nearby']].sort_values(
+                user_datetime
+            )
+
+            move_data = df_event[df_event[event_id_] == event_id]
+
+            base_map = add_event_folium(
+                move_data,
+                event_lat=event_lat_,
+                event_lon=event_lon_,
+                event_point=event_point,
+                slice_tags=slice_event_show
+            )
+
+            # keep only the first tnz_point nearby to cvp for each tnz
+            df_begin.drop_duplicates(
+                subset=[user_id, 'nearby'],
+                inplace=True
+            )
+
+            # for each tnz nearby to cvp
+            tnzs = []
+
+            # create an empty dataframe
+            columns_names = list(move_data.columns)
```

Comment:
Codacy found an issue: [local variable 'columns_names' is assigned to but never used (F841)](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=5714604)